crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -1359,10 +1359,7 @@ pub(crate) fn build_transaction_receipt_with_block_receipts(
 
     #[cfg(feature = "optimism")]
     {
-        let mut op_fields = OptimismTransactionReceiptFields {
-            deposit_nonce: receipt.deposit_nonce.map(U64::from),
-            ..Default::default()
-        };
+        let mut op_fields = OptimismTransactionReceiptFields::default();
 
         if let Some(l1_block_info) = optimism_tx_meta.l1_block_info {
```

Comment:
I meant keeping the if deposit check as the outer predicate but just flip the the arms so we don't need `!`